File: Ruby/Microverse/sorting/insertion/insertion_sort.rb
def insertion_sort(array)
  insertions = 0
  len = array.length - 1
  indices = [*1..len]
  indices.each do |idx|
    key = array[idx]
    j = idx - 1
    while j >= 0 && array[j] > key
      array[j + 1] = array[j]
      j -= 1
    end
    array[j + 1] = key
    p "Insert => #{array}"
    insertions += 1
  end
  array
end

p insertion_sort([2, 1, 3, 1, 2])
p insertion_sort([1, 4, 3, 6, 9, 2])

File: Python/algorithms/sorting/insertion_sort.py
def insertion_sort(array):
    for idx in range(1, len(array) - 1):
        key = array[idx]
        moving_idx = idx - 1
        while moving_idx >= 0 and array[moving_idx] > key:
            array[moving_idx + 1] = array[moving_idx]
            moving_idx -= 1
        array[moving_idx + 1] = key
        print(f'Insert No. {idx} => {array}')
    return array


print(insertion_sort([1, 4, 3, 6, 9, 2]))
# => 1 4 3 6 9 2
#    1 3 4 6 9 2
#    1 3 4 6 9 2
#    1 3 4 6 9 2
#    1 2 3 4 6 9